writeToStream.js

```diff
@@ -520,7 +521,7 @@ function writeString (stream, string) {
  * @api private
  */
 function writeNumber (stream, number) {
-  return stream.write(numCache[number])
+  return stream.write(numCache.get(number, generate.cacheNumbers))
 }
```

Comment:
I think we can have two functions, `writeNumberCached` and  `writeNumberAllocate`, and we can assign them to `writeNumber` in the `cacheNumbers` setter.